src/spec.c

```diff
@@ -822,6 +835,34 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, ArgsCursor *ac, QueryError
         }
       }
       fs->ftId = textId;
+      if isSpecJson (sp) {
+        if ((sp->flags & Index_HasFieldAlias) && (sp->flags & Index_StoreTermOffsets)) {
+          RedisModuleString *err_msg;
+          JSONPath jsonPath = pathParse(fs->path, &err_msg);
```

Comment:
If we get `err_msg` here, shouldn't we return right away with error?

Reply:
Failure could be due to an erroneous path, or since the required API is missing.
Upon real failure (`jsonPath` is NULL and `err_msg` is non-NULL)
We return right after setting the error status and freeing `err_msg`.
Otherwise, we continue with the rest of the fields.